nps_active_space/utils/models.py

```diff
@@ -178,6 +183,231 @@ def _validate(self, columns: List[str]):
+                df = df.drop(columns = ['tslc', 'dur_secs', 'diff_flight', 'cumsum', 'valid_BARO', 'valid_VERTICAL_VELOCITY', 'SIMULATED_REPORT', 'valid_IDENT', 'valid_CALLSIGN', 'valid_VELOCITY', 'valid_HEADING', 'valid_ALTITUDE', 'valid_LATLON', 'DATE'])
+
+                data = data.append(df)
+
+            data = gpd.GeoDataFrame(
+                data,
+                geometry=gpd.points_from_xy(data["lon"], data["lat"]),
+                crs="epsg:4326"
+            )
+
+        return data
+
+
+class EarlyAdsb(Adsb):
+
+    """
+    A geopandas GeoDataFrame wrapper class to ensure consistent ADS-B data.
+
+    Parameters
+    ----------
+    filepaths_or_data : List, str, or gpd.GeoDataFrame
+        A directory containing ADS-B TSV files, a list of ADS-B TSV files, or an existing gpd.GeoDataFrame of ADS-B data.
+    """
+
+    def __init__(self, filepaths_or_data: Union[List[str], str, gpd.GeoDataFrame]):
+        data = self._read(filepaths_or_data)
+        data.drop_duplicates(subset=['TIME'], inplace=True, keep = 'last')
+        super().__init__(data=data)
+
+    def _read(self, filepaths_or_data: Union[List[str], str, gpd.GeoDataFrame]):
+        """
+        Read in ADS-B points as formatted by early-development NPS data loggers (circa 2019).
+
+        Parameters
+        ----------
+        filepaths_or_data : List, str, or gpd.GeoDataFrame
+            A directory containing ADS-B files, a list of ADS-B files, or an existing gpd.GeoDataFrame of ADS-B data.
+
+        Raises
+        ------
+        AssertionError if directory path or file path does not exists or is of the wrong format.
+        """
+        if isinstance(filepaths_or_data, gpd.GeoDataFrame):
+            #self._validate(filepaths_or_data.columns)
+            data = filepaths_or_data.to_crs("epsg:4326")
+
+        else:
+            if isinstance(filepaths_or_data, str):
+                assert os.path.isdir(filepaths_or_data), f"{filepaths_or_data} does not exist."
+                filepaths_or_data = glob.glob(f"{filepaths_or_data}/*.TSV")
+
+            else:
+                for file in filepaths_or_data:
+                    assert os.path.isfile(file), f"{file} does not exist."
+                    assert (file.endswith('.txt')|file.endswith('.TSV')), f"Only .TSV ADS-B files accepted."
+
+            data = pd.DataFrame()
+            for file in tqdm(filepaths_or_data, desc='Loading ADS-B files', unit='files', colour='green'):
+                df = pd.read_csv(file, sep="\t")
+
+                df.columns = ["ICAO_address", "TIME", "lat", "lon", "altitude"]
+                df["TIME"] = df["TIME"].apply(lambda t: dt.datetime.strptime(t, "%Y/%m/%d %H:%M:%S.%f").replace(microsecond=0))
+                df["DATE"] = df["TIME"].dt.strftime("%Y%m%d")
+
+                # Sort records by ICAO Address and TIME then reset dataframe index
+                df.sort_values(["ICAO_address", "TIME"], inplace=True)
```

Comment:
You can do these two lines in one line by adding `ignore_index=True` to the sort_values call